### ISSUE TITLE
Display the most recent version number of plugins and themes with available updates

### ISSUE THREAD

**Issue by baizmandesign** [wp-cli/ideas#174]
## Feature Request

- [X] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).

**Describe your use case and the problem you are facing**

When a plugin or theme has an available update, the WordPress dashboard will display a link to update to version X.X.X. With WP CLI, when an update for a plugin is available, the version number isn't displayed. For example:

```
$ wp plugin list                                                                          
+---------------------------------------------+----------+-----------+---------+
| name                                        | status   | update    | version |
+---------------------------------------------+----------+-----------+---------+
| advanced-custom-fields-pro                  | active   | available | 6.0.7   |
+---------------------------------------------+----------+-----------+---------+
```

The command only reveals that an update is "available," but we don't see the new version number. This also affects themes. For example:

```
$ wp theme list
+------------------------+--------+-----------+---------+
| name                   | status | update    | version |
+------------------------+--------+-----------+---------+
| twentytwentyone        | parent | available | 1.7     |
+------------------------+--------+-----------+---------+
```

It would be nice if the newly available plugin and theme versions appeared in both the `wp plugin list` / `wp plugin status` screens and `wp theme list` / `wp theme status` screens.

Why is this useful or important? For one thing, I think the CLI should mirror the GUI (at least in this respect). But the real value is the same as what can be gleaned from seeing the version number in the GUI: the ability to quickly calculate how many versions behind a plugin is, or in plain English, how out-of-date it is.

**Describe the solution you'd like**

I'm not sure it's as simple as the screen below:

```
$ wp plugin list                                                                          
+---------------------------------------------+----------+-----------+---------+
| name                                        | status   | update    | version |
+---------------------------------------------+----------+-----------+---------+
| advanced-custom-fields-pro                  | active   | 6.1.2     | 6.0.7   |
+---------------------------------------------+----------+-----------+---------+
```

(A large part of me wants to swap the third and fourth columns.)

The status screens would also need to be revisited.

This breaks the command `wp plugin list --update=available`. Maybe the update parameter would take a wildcard, like `--update="*"` for any plugin version (that has an update) or `--update="5.*"` for any plugin with a version of 5 (that has an update). Perhaps "available" is translated to "*" for backward compatability?

The `wp plugin status` page adds green text saying "(Update available)." Maybe the new version number is in green on the above screen so that it stands out?

**Sidenote:** the version information is available in the `wp_options` table. The `option_name` for plugin data is `_site_transient_update_plugins` and for theme data is `_site_transient_update_themes`.

**Comment by baizmandesign** [comment]
Hey, @danielbachhuber. Have you had a chance to take a look at this? I'd love to know your thoughts.

**Comment by danielbachhuber** [comment]
@baizmandesign Sorry, have been focusing elsewhere this week.


> Why is this useful or important? For one thing, I think the CLI should mirror the GUI (at least in this respect).

I like this idea. @porg [requested something similar](https://github.com/wp-cli/extension-command/issues/344#issuecomment-1437408322), and I'm open to the idea that the CLI should mirror the GUI.

> This breaks the command `wp plugin list --update=available`. Maybe the update parameter would take a wildcard, like `--update="*"` for any plugin version (that has an update) or `--update="5.*"` for any plugin with a version of 5 (that has an update). Perhaps "available" is translated to "*" for backward compatability?

The update version is available in the `update_version` field. We should use this, instead of breaking backward compatibility.

I'm open to looking at a pull request if someone would like to submit one.

**Comment by baizmandesign** [comment]
Thanks for your response, @danielbachhuber. I can take a crack at this. Feel free to assign it to me.

I must admit I don't understand the comment about the update version being available "in the `update_version` field." Where is this? In the transient in the database?

And what of the wildcard idea? Keep or toss?

**Comment by danielbachhuber** [comment]
> I can take a crack at this. Feel free to assign it to me.

@baizmandesign Sounds great!

> I must admit I don't understand the comment about the update version being available "in the `update_version` field." Where is this? In the transient in the database?

Sorry, to clarify:

```
wp plugin list --fields=name,update_version,update,version
+---------------------+----------------+-----------+---------+
| name                | update_version | update    | version |
+---------------------+----------------+-----------+---------+
| akismet             |                | none      | 5.1     |
| elementor           |                | none      | 3.12.1  |
| gutenberg           |                | none      | 15.5.1  |
| hello               |                | none      | 1.7.2   |
| servebolt-optimizer |                | none      | 3.5.18  |
| wordpress-importer  |                | none      | 0.8     |
| wp-redis            | 1.3.5          | available | 1.3.4   |
+---------------------+----------------+-----------+---------+
```

> And what of the wildcard idea? Keep or toss?

Can you share an example of how you might use the wildcard in your daily work?

**Comment by baizmandesign** [comment]
Ah, there `update_version` is! Thank you!

I didn't realize that one could request this field and that it would provide the information I was seeking. That changes the nature of my request. I suppose I'm advocating for combining the `update` and `update_version` fields. Your comment to display the version information from the `update_version` field in the `update` field solidifies this line of thinking. *(Why have two fields displaying virtually the same information?)* But I'm not sure combining them—and possibly eliminating the no-longer-useful `update_version` field—is the right approach since it may break backward compatibility. Maybe it is retained for the time being but gets phased out in a future release?

How do you wish to proceed? Keep the two fields with redundant information, eliminate a field, or some third unimagined option?

**Comment by danielbachhuber** [comment]
I don't think it makes sense to break backwards compatibility on the fields. We could display more fields by default, though.

**Comment by baizmandesign** [comment]
@danielbachhuber Not breaking backwards compatibility sounds good to me.

So at this point, I'm confused about what changes I should make to WP CLI. Are we just going to add the `update_version` field to the default output of `wp plugin list`? This seems... verbose to me, but I can do it. I think, in the long run, I'd like to see the `update_version` field replace the `update` field in the default output of `wp plugin list` (and the theme equivalent).

**Comment by baizmandesign** [comment]
Hey, @danielbachhuber. After further thought, I think we can close this issue without any further action. The default behavior that I suggested looks like it won't come to pass (which is fine, because of the next point), and I'm able to display the information that I want to see by customizing `~/.wp-cli/config.yml` like so:

```
plugin list:
     fields: name,version,update_version
theme list:
     fields: name,version,update_version
```

Thanks for hearing me out! I appreciate your feedback and guidance.

**Comment by danielbachhuber** [comment]
@baizmandesign Sounds good! Thanks for letting me know 😊
